.github/workflows/publish.yml

```
@@ -8,48 +8,26 @@ permissions:
   id-token: write
   contents: read
 
-env:
-  POETRY_VERSION: "1.6.1"
-  PYTHON_VERSION: "3.11"
-
 jobs:
   publish:
     environment:
       name: pypi
       url: https://pypi.org/p/ziggy-pydust-template
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        id: setup-python
-        with:
-          python-version: ${{ env.PYTHON_VERSION }}
+      - name: 🛎️ Checkout
+        uses: actions/checkout@v4
 
-      - name: Load cached Poetry installation
-        id: cached-poetry
-        uses: actions/cache@v4
-        with:
-          path: ~/.local
-          key: poetry-${{ runner.os }}-${{ env.POETRY_VERSION  }}-${{ steps.setup-python.outputs.python-version }}
-      - name: Install Poetry
-        id: install-poetry
-        if: steps.cached-poetry.outputs.cache-hit != 'true'
-        uses: snok/install-poetry@v1
-        with:
-          version: ${{ env.POETRY_VERSION }}
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-      - name: Poetry git plugin
-        run: poetry self add poetry-git-version-plugin
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v4
+      - name: 📦 Install uv
+        uses: astral-sh/setup-uv@v4
```

see my comment above for the juicy deets:

```suggestion
        uses: astral-sh/setup-uv@v4
        with:
          python-version: "3.12"
```

---

v5 and enable cache https://github.com/spiraldb/actions/blob/develop/.github/actions/setup-uv/action.yml#L21 maybe